update-gh-pages.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+#
+# This script will build the API reference docs, copy them to the gh-pages
+# branch, and tell you whether you need to commit changes.
+
+set -e -o pipefail
+
+CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+: ${CURRENT_BRANCH:=master}
+
+(
+  set -x;
+  ./gradlew dokka && \
+  git checkout gh-pages && \
+  git pull && \
+  rm -Rf *.css grpc-kotlin && \
+  cp -R build/dokka/* .
+)
+
+if [[ -z $(git status --porcelain) ]]; then
+  echo
+  echo "No updates necessary to the API reference."
+  if [[ -n "$CURRENT_BRANCH" ]]; then
+    echo "Switching back to branch $CURRENT_BRANCH"
+    (set -x; git checkout $CURRENT_BRANCH)
+  fi
+else
+  echo
+  echo "IMPORTANT:"
+  echo "  You're on the gh-pages branch now."
+  echo "  Pages of the API reference have changed; for details, run: git status"
+  echo "  You need to commit and push the changes to the upstream gh-pages branch."
+  exit 1
```

Comment:
So this script doesn't actually publish the changes, instead it exits with an error code if changes exist?

A better workflow would be it pushes the changes if they exist. That way we can run this in `.travis.yml` for the master branch only. Something like this:

```sh
else
    git add .
    git ci -am "Update reference docs from [SHA]"
    git push origin gh-pages
fi
```

Reply:
As discussed, I've added a `--push` option.